crates/fj-kernel/src/objects/full/edge.rs

```diff
@@ -86,6 +86,16 @@ impl HalfEdge {
         self.curve.point_from_path_coords(start)
     }
 
+    /// Compute the surface position where the half-edge ends
+    pub fn end_position(&self) -> Point<2> {
+        // Computing the surface position from the curve position is fine.
+        // `HalfEdge` "owns" its end position. There is no competing code that
+        // could compute the surface position from slightly different data.
```

Comment:
This comment is completely false, and dangerously so. It used to be true for the *start* position (from where it was adapted), but by adding this method, you add the competing code that the comments in both methods claim doesn't exist.

This is a problem, because sooner or later someone is going to call this method to compare it to the start position of the next half-edge in the cycle, or use it in some scenario where it's assumed that both positions are going to be equal.

I think this method shouldn't exist. It's still possible to make the same mistake doing exactly what the method does internally, but at the very least we shouldn't make that easy. (And there should be better documentation around this, but that's a separate issue.)

However, what the new validation check does is perfectly fine, as it uses an epsilon value (`config.identical_max_distance`) for the comparison. I think it would be best to inline the body of this method into the validation check.